Software/Code/Code/Core/Inc/main.h

```diff
@@ -0,0 +1,134 @@
+/* USER CODE BEGIN Header */
+/**
+  ******************************************************************************
+  * @file           : main.h
+  * @brief          : Header for main.c file.
+  *                   This file contains the common defines of the application.
+  ******************************************************************************
+  * @attention
+  *
+  * Copyright (c) 2024 STMicroelectronics.
+  * All rights reserved.
+  *
+  * This software is licensed under terms that can be found in the LICENSE file
+  * in the root directory of this software component.
+  * If no LICENSE file comes with this software, it is provided AS-IS.
+  *
+  ******************************************************************************
+  */
+/* USER CODE END Header */
+
+/* Define to prevent recursive inclusion -------------------------------------*/
+#ifndef __MAIN_H
+#define __MAIN_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Includes ------------------------------------------------------------------*/
+#include "stm32h7xx_hal.h"
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+#include <stdbool.h>
+#include <stdint.h>
+/* USER CODE END Includes */
+
+/* Exported types ------------------------------------------------------------*/
+/* USER CODE BEGIN ET */
+
+enum Direction_E
+{
+   DIRECTION_FORWARD_MIN,
+   DIRECTION_FORWARD_NOMINAL,
+   DIRECTION_FORWARD_MAX,
+   DIRECTION_REVERSE_MIN,
+   DIRECTION_REVERSE_NOMINAL,
+   DIRECTION_REVERSE_MAX,
+   DIRECTION_ERROR,
+   DIRECTION_NOT_AVAILABLE
+};
+
+typedef float    Hz_T;
+typedef uint32_t us_T;
```

Comment:
Move to an application header file instead. This does not need to be shared. Likewise with the exported constants user section below and anywhere else applicable in this file.